py/mattak/backends/uproot/voltage_calibration.py

```diff
@@ -176,6 +177,38 @@ def __call__(self, waveform_array : numpy.ndarray, starting_window : Union[float
             return calibrate(waveform_array, self.__cal_param, [self.__adc_table_voltage, self.__adc_table_voltage],
                              self.__cal_residuals_adc, starting_window, upsampling=False)  # already upsampled
 
+    def plot_ch(self, xs=numpy.linspace(-1000, 1000, 100), ch=0):
+
+        from matplotlib import pyplot as plt
+
+        fig, (ax, ax2) = plt.subplots(2, 1, height_ratios=[3, 1], sharex=True)
+
+        tables = numpy.array([self.__get_adc_table(ch, i) for i in range(self.NUM_DIGI_SAMPLES)])
+        out = []
+        out2 = []
+        for x in xs:
+            # vs = [numpy.interp(x, t, self.__adc_table_voltage) * 1000 for t in tables]
+            # out.append([numpy.mean(vs), numpy.std(vs)])
+            vs2 = self.calibrate([numpy.full(self.NUM_WF_SAMPLES, x)], [0], channels=[ch]) * 1000
+            out2.append([numpy.mean(vs2), numpy.std(vs2)])
+
+        out = numpy.array(out2)
+
+        ax.errorbar(xs, out[:, 0], out[:, 1], ls="", label=f"Ch {ch} (mean over all samples)")
+        ax.plot(xs, xs * 2500 / 4096, "k--", lw=1, label="2500 / 4096")
+        ax2.set_xlabel("input ADC")
+        ax.set_ylabel("output voltage / mV")
+
+        ax2.errorbar(xs, out[:, 0] - xs * 2500 / 4096, out[:, 1], ls="", label=f"Ch {ch}")
+        ax2.set_ylabel("residual / mV")
+
+        ax.grid()
+        ax2.grid()
+        ax.legend()
+        fig.tight_layout()
+        name = self.__path.replace('.root', f'_ch{ch}_test.png')
+        plt.savefig(name, transparent=False)
```

Comment:
I guess this is never meant to be used interactively? Suggest that output location could be changed? (in the case of loading VC from a central place, may not have write permissions anyway)

Reply:
Can do

Reply:
What do mean with interactively? This is somewhat for debugging

Reply:
interactively as in popping open a window (plt.show () rather than plt.savefig() ).

Reply:
Ah, I see. We could do that as well...

Reply:
switched to show since it is meant for debug.

Reply:
The plots are actually interesting!